### PR TITLE
MM-48137 "Join/Leave" notification refactor

### PR DESCRIPTION
**Pull request by levb**
Need to summarize

### CODE REVIEW

File: server/appservices/subscriptions.go [hunk 1]
```diff
@@ -38,6 +38,34 @@ func (a *AppServices) Subscribe(r *incoming.Request, sub apps.Subscription) erro
 		return err
 	}
 
+	// TODO <>/<> MM-??? deprecate bot_joined|left_...
```
Comment (javaguirre) [inline server/appservices/subscriptions.go:41]: I guess we need to create a ticket for this and remove the comment?

File: server/appservices/subscriptions.go [hunk 2]
```diff
@@ -166,6 +194,17 @@ func (a *AppServices) unsubscribe(r *incoming.Request, ownerUserID string, e app
 		return modified, nil
 	}
 
+	// TODO <>/<> MM-??? deprecate bot_joined|left_...
```
Comment (javaguirre) [inline server/appservices/subscriptions.go:197]: ditto

File: test/app/subscriptions.go [hunk 3]
```diff
@@ -39,7 +39,27 @@ var allSubjects = map[apps.Subject]apps.Expand{
...
+		User:       apps.ExpandAll,
+		Team:       apps.ExpandAll,
+		TeamMember: apps.ExpandAll,
+	},
+	apps.SubjectSelfLeftTeam: {
+		User:       apps.ExpandAll,
+		Team:       apps.ExpandAll,
+		TeamMember: apps.ExpandAll,
+	},
+	// apps.SubjectSelfMentioned: {
```
Comment (javaguirre) [inline test/app/subscriptions.go:62]: could we remove the code? we can recover it with `git log -p`

**Review by javaguirre — CHANGES_REQUESTED** [review]
LGTM, only a couple of comments.

File: server/appservices/subscriptions.go (hunk 1 above)
Comment (levb) [reply server/appservices/subscriptions.go:41]: yes, I actually want to sprinkle the core with the ticket reference to make it easier to remove later.

File: test/app/subscriptions.go (hunk 3 above)
Comment (levb) [reply test/app/subscriptions.go:62]: @mickmister @DHaussermann I am maintaining my view that functionality that is covered by restapitest should be excluded from the test app to minimize its complexity and maintenance. Why do we need subscriptions in the test app?

**Comment by levb** [comment]
Going to combine the 2 subs PRs, clean up and request a fresh review

**Pull request merged**